spring-cloud-open-service-broker-core/src/main/java/org/springframework/cloud/servicebroker/controller/ServiceInstanceController.java

```diff
@@ -241,9 +241,18 @@ else if (response.isInstanceExisted()) {
 						.doOnError(e -> LOG.error("Error getting service instance last operation. error=" +
 								e.getMessage(), e)))
 				.map(response -> {
-					boolean isSuccessfulDelete = response.getState().equals(OperationState.SUCCEEDED) && response
+					boolean isSuccessfulDelete = OperationState.SUCCEEDED.equals(response.getState()) && response
 							.isDeleteOperation();
 					return new ResponseEntity<>(response, isSuccessfulDelete ? HttpStatus.GONE : HttpStatus.OK);
+				})
+				.onErrorResume(e -> {
+					if (e instanceof ServiceInstanceDoesNotExistException) {
+						// TODO: v2.16 of the OSB API spec changes this to an HTTP 404
+						return Mono.just(new ResponseEntity<>(HttpStatus.BAD_REQUEST));
```

Comment:
I haven't checked the OSB spec yet, but it would be nice if we can add response body to explain the error here.

Reply:
That's a good idea. There's a `description` field that I can use for that. Thanks!

Reply:
done. updated to include a description.